fili-core/src/test/groovy/com/yahoo/bard/webservice/async/AsyncDruidSendsErrorSpec.groovy

```diff
@@ -75,6 +77,33 @@ class AsyncDruidSendsErrorSpec extends AsyncFunctionalSpec {
                         "druidQuery" : null
                     }"""
 
+    final CountDownLatch jobMetadataReady = new CountDownLatch(1)
+
+    def setup() {
+        TestAsynchronousWorkflowsBuilder.addSubscriber(
+                TestAsynchronousWorkflowsBuilder.Workflow.JOB_MARKED_COMPLETE,
+                new Observer() {
```

Comment:
This is ugly. We can do better (after all, this _is_ Groovy):

```groovy
TestAsynchronousWorkflowsBuilder.addSubscriber(TestAsynchronousWorkflowsBuilder.Workflow.JOB_MARKED_COMPLETE) 
        { jobMetadataReady.countDown() }
        { throw it }
```
Or we can pass them as parameters if you don't like the implicit closure-passing style

```groovy
TestAsynchronousWorkflowsBuilder.addSubscriber(
        TestAsynchronousWorkflowsBuilder.Workflow.JOB_MARKED_COMPLETE,
        {jobMetadataReady.countDown()},
        {throw it}
)
```

To achieve this, we need to add some more helpers to the `TestAsynchronousWorkflowsBuilder`:

```java
/**
 * Adds the specified subscriber to the specified workflow.
 *
 * @param workflow  The workflow to add the countdown latch to
 * @param workflowSubscriber  The subscriber that should be added to the specified workflow
 */
public static void addSubscriber(Workflow workflow, Observer workflowSubscriber) {
    SUBSCRIBERS.put(workflow, workflowSubscriber);
}

/**
 * Adds the subscriber (specified in individual method components) to the specified workflow.
 *
 * @param workflow  The workflow to add the countdown latch to
 * @param onNext  onNext method for the observer
 * @param onCompleted  onCompleted method for the observer
 * @param onError  onError method for the observer
 */
public static void addSubscriber(
        Workflow workflow,
        Consumer<Object> onNext,
        Runnable onCompleted,
        Consumer<Throwable> onError
) {
    Observer workflowSubscriber = new Observer() {
        @Override
        public void onNext(Object next) {
            onNext.accept(next);
        }
        @Override
        public void onCompleted() {
            onCompleted.run();
        }
        @Override
        public void onError(Throwable error) {
            onError.accept(error);
        }
    };
    SUBSCRIBERS.put(workflow, workflowSubscriber);
}

/**
 * Subscribes the partially-specified observer to the specified workflow.
 * <p>
 * Uses a no-op onCompleted method.
 *
 * @param workflow  The workflow to add the countdown latch to
 * @param onNext  onNext method for the observer
 * @param onError  onError method for the observer
 */
public static void addSubscriber(Workflow workflow, Consumer<Object> onNext, Consumer<Throwable> onError) {
    addSubscriber(workflow, onNext::accept, () -> { }, ignored -> { });
}

/**
 * Subscribes the partially-specified observer to the specified workflow.
 * <p>
 * Uses no-op onCompleted and onError methods.
 *
 * @param workflow  The workflow to add the countdown latch to
 * @param onNext  onNext method for the observer
 */
public static void addSubscriber(Workflow workflow, Consumer<Object> onNext) {
    addSubscriber(workflow, onNext::accept, ignored -> { });
}
```